git_greppers/utils.sh

```
@@ -0,0 +1,127 @@
+}
+
+# Extract repository identifier from URL
+extract_repo_identifier() {
+  local repo_type=$1
+  local repo_url=$2
+  
+  # Send debug output to stderr
+  debug 3 "Extracting repository identifier from URL: $repo_url for type: $repo_type" >&2
+  
+  # Remove .git suffix if present and transform SSH URL to HTTPS format if needed
+  local url=$(echo $repo_url | sed -e 's/\.git$//' -e 's/git@github.com:/https:\/\/github.com\//' -e 's/git@gitlab.com:/https:\/\/gitlab.com\//')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)